components/push/src/lib.rs

```diff
@@ -380,17 +380,18 @@ impl PushManager {
         encoding: &str,
         salt: &str,
         dh: &str,
-    ) -> Result<Vec<u8>> {
-        // TODO(teshaq): Modify the decrypt function to return the Vec<u8> directly
-        // once the ffi crate is no longer using it
-        let ret = self.internal.lock().unwrap().decrypt(
+    ) -> Result<Vec<i8>> {
+        let decrypted = self.internal.lock().unwrap().decrypt(
             channel_id,
             body,
             encoding,
             Some(salt),
             Some(dh),
         )?;
-        Ok(ret.as_bytes().to_vec())
+
+        // NOTE: this returns a `Vec<i8>` since the kotlin consumer is expecting
+        // signed bytes.
+        Ok(decrypted.into_iter().map(|ub| ub as i8).collect())
```

Comment:
Is this related to unsigned types being experimental in Kotlin? If so, I think that's no longer the case.  (IOW, if it's possible to now use u8 here I think we should do so, but I've no idea how practical that is, so feel free to ignore!)

Reply:
Uhh yes, I used this because Android components expects a `ByteArray`, which is just one `.toByteArray()` away from a `List<Byte>`, but `List<UByte>` can't convert directly. So A-C will have to convert the `List<UByte>` to `ByteArray` through a `UByteArray`... which isn't _that_ bad, but it would be the first time they interact with unsigned types over there 😅 I guess it's just a matter of do we want to do the conversion for them, or have them do the conversion themselves